plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -67,6 +69,8 @@
 public class VacuumProcedure
         implements Provider<Procedure>
 {
+    private static final int MAX_SUPPORTED_WRITER_VERSION = 3;
+    private static final int MAX_SUPPORTED_READER_VERSION = 2;
```

Comment:
I've made these separate because I think we can advance them faster than we advance support for updates/deletes. As long as newer versions don't include new types of metadata files we can vacuum them.